### PR TITLE
Expose kube-controllers prometheus metrics

### PR DESCRIPTION
**Pull request by caseydavenport**
## Description

These were added in Calico v3.18, will be present in Calico Enterprise v3.6.

This PR enables scraping of these metrics by default in Calico
Enterprise.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/kube-controllers.go [hunk 1]
```diff
@@ -423,6 +428,34 @@ func (c *kubeControllersComponent) controllersDeployment() *apps.Deployment {
...
+			Name:      "calico-kube-controllers-metrics",
+			Namespace: common.CalicoNamespace,
+			Labels:    map[string]string{"k8s-app": "calico-kube-controllers"},
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{"k8s-app": "calico-kube-controllers"},
+			Type:     v1.ServiceTypeClusterIP,
+			Ports: []v1.ServicePort{
+				// Expose on the default port - 9094.
+				// TODO: Users can change this value in KubeControllersConfiguration. We should
```
Comment (tmjd) [inline pkg/render/kube-controllers.go:446]: Wouldn't it be relatively easy to watch KubeControllersConfiguration and read this field. Is there a good reason to not do that yet?

File: pkg/render/kube-controllers.go [hunk 2]
```diff
@@ -423,6 +434,31 @@ func (c *kubeControllersComponent) controllersDeployment() *apps.Deployment {
...
+			Labels:    map[string]string{"k8s-app": "calico-kube-controllers"},
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{"k8s-app": "calico-kube-controllers"},
+			Type:     v1.ServiceTypeClusterIP,
+			Ports: []v1.ServicePort{
+				{
+					Name:       "metrics-port",
+					Port:       int32(c.metricsPort),
+					TargetPort: intstr.FromInt(int(9094)),
```
Comment (tmjd) [inline pkg/render/kube-controllers.go:454]: Isn't the TargetPort the one that would need to change?
I'm not sure if we should set both to the values configured or just the TargetPort, probably just the TargetPort but whatever you think is best.

**Review by tmjd — CHANGES_REQUESTED** [review]
I think one issue.
This will probably require a ClusterRole update to the docs.

**Comment by caseydavenport** [comment]
@tmjd ready for a final look!

> This will probably require a ClusterRole update to the docs.

Yep, definitely. Have submitted PR here for that: https://github.com/projectcalico/calico/pull/4438

**Review by tmjd — APPROVED** [review]
LGTM

**Comment by tmjd** [comment]
Though it seems like several tests are failing that will need to be fixed.

**Pull request merged**